mentorship ios/ViewModels/RelationViewModel.swift

```diff
@@ -27,72 +27,23 @@ class RelationViewModel: ObservableObject {
-        firstTimeLoad = false
-    }
-    
-    func fetchCurrentRelation() {
-        //get auth token
-        guard let token = try? KeychainManager.getToken() else {
-            return
-        }
-        
-        //set inActivity
-        inActivity = true
-        
-        //api call
-        cancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.currentRelation, token: token)
-            .receive(on: RunLoop.main)
-            .catch { _ in Just(self.currentRelation) }
-            .sink { [weak self] current in
-                //use current relation data
-                self?.inActivity = false
-                self?.currentRelation = current
-                self?.personName = self?.getPersonNameAndType(data: current) ?? ""
-                //chain api call. get current tasks using id from current relation
-                //if current relation invalid, delete all tasks and return
-                guard let currentID = current.id else {
-                    self?.toDoTasks.removeAll()
-                    self?.doneTasks.removeAll()
-                    return
-                }
-                self?.fetchTasks(id: currentID, token: token)
-        }
-    }
-    
-    func fetchTasks(id: Int, token: String) {
-        tasksCancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.getCurrentTasks(id: id), token: token)
-            .receive(on: RunLoop.main)
-            .catch { _ in Just(self.tasks) }
-            .sink { [weak self] tasks in
-                //if success
-                if NetworkManager.responseCode == 200 {
-                    self?.doneTasks.removeAll()
-                    self?.toDoTasks.removeAll()
-                    //assign done tasks and tasks to do
-                    for task in tasks {
-                        if task.isDone ?? false {
-                            self?.doneTasks.append(task)
-                        } else {
-                            self?.toDoTasks.append(task)
-                        }
-                    }
-                }
-                //else show fail alert
-                else {
-                    self?.showErrorAlert = true
-                    self?.alertMessage = LocalizableStringConstants.operationFail
-                }
+    func handleFetchedTasks(tasks: [TaskStructure], success: Bool) {
+        if success {
+            doneTasks.removeAll()
+            toDoTasks.removeAll()
+            for task in tasks {
+                task.mapTo(viewModel: self)
+            }
+        } else {
+            showErrorAlert = true
+            alertMessage = LocalizableStringConstants.operationFail
         }
     }
     
     //func to get name of other person in current relation.
-    func getPersonNameAndType(data: HomeModel.HomeResponseData.RequestStructure) -> String {
+    func getPersonNameAndType(data: RequestStructure) -> String {
```

Comment:
you need this method because your ViewModel does not hold correct data in correct format.

Reply:
Fixed.